mfdeveloper/UnityAccessibilityPlugin
# Unity official translation support: UPM package integration

## Description

Added support to custom external plugins translations, in order to translate keys for **Speech Output** for any `UAP_BaseElement`, and **Group Name** for `AccessibleUIGroupRoot`. 
To acomplish that, now you can pass a custom **Localization Strategy** ScriptableObject asset to `Accessibility Manager` singleton gameObject.

Besides that, was added a "_monorepo_" structure with 2 packages:

- **`com.metalpopgames.uiaccessibility`**: Original package
- **`com.metalpopgames.uiaccessibility.unity-localization`**: Unity translation support. This additional package that use `com.unity.localization` official UPM package as dependency

## How to test

1. Select the `Accessibility Manager` singleton gameObject from scene hierarchy => **Settings** tab, and attach a custom `ScriptableObject` strategy that implements a integration with any external translation plugin
   > For instance, use `UnityLocalizationStrategy` from sub-package: **`com.metalpopgames.uiaccessibility.unity-localization`**)

   
    ![unity-editor-uap-translation-plugin](https://github.com/mfdeveloper/UnityAccessibilityPlugin/assets/431606/97f400c0-ad37-473c-999e-53895af4e2a0)


2. After that, attach the script component `Localize String Event` to the Acessible Element that needs be translated (e.g an `AccessibleButton`), and add a string key. Here we are using **`DisabledButton`**.

   ![unity-editor-uap-translation-attach-localize-string-event-script](https://github.com/mfdeveloper/UnityAccessibilityPlugin/assets/431606/e3263e93-4daa-41ae-8500-d142e515e4a3)

3. Now, select `Accessible Button` component => **Speech Output**, and mark the 2 checkboxes: `Custom Hint Text` and `Is Localization Key`. The hint text field should be updated with the same key added before.

   ![unity-editor-uap-accessible-button-translation](https://github.com/mfdeveloper/UnityAccessibilityPlugin/assets/431606/3df41d31-46dd-48ba-b0f8-49a81c86ff5b)

### Custom Hint: when a element is DISABLED

Optionally, configure to say the custom hint text only when the element is **DISABLED**

![image](https://github.com/mfdeveloper/UnityAccessibilityPlugin/assets/431606/6f0b2c78-8dbd-4257-9373-afc67b0c8a7d)
